BUG: incompatibility with pytest 8.0dev

New failures due to https://github.com/pytest-dev/pytest/pull/11268, but I haven't yet looked into solutions yet.


https://github.com/scientific-python/pytest-doctestplus/actions/runs/5814842359

```
 ==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
../../.tox/py312-test-pytestdev/lib/python3.12/site-packages/_pytest/runner.py:340: in from_call
    result: Optional[TResult] = func()
../../.tox/py312-test-pytestdev/lib/python3.12/site-packages/_pytest/runner.py:371: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
../../.tox/py312-test-pytestdev/lib/python3.12/site-packages/_pytest/main.py:756: in collect
    for x in self._collectfile(path):
../../.tox/py312-test-pytestdev/lib/python3.12/site-packages/_pytest/main.py:605: in _collectfile
    if ihook.pytest_ignore_collect(collection_path=fspath, config=self.config):
../../.tox/py312-test-pytestdev/lib/python3.12/site-packages/_pytest/config/compat.py:66: in fixed_hook
    return hook(**kw)
../../.tox/py312-test-pytestdev/lib/python3.12/site-packages/pluggy/_hooks.py:433: in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
../../.tox/py312-test-pytestdev/lib/python3.12/site-packages/pluggy/_manager.py:112: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
../../.tox/py312-test-pytestdev/lib/python3.12/site-packages/pytest_doctestplus/plugin.py:512: in pytest_ignore_collect
    collect_ignore = config._getconftest_pathlist("collect_ignore",
E   TypeError: Config._getconftest_pathlist() got an unexpected keyword argument 'rootpath'
=========================== short test summary info ============================
```

--- a/pytest_doctestplus/plugin.py
+++ b/pytest_doctestplus/plugin.py
@@ -26,9 +26,10 @@ from .output_checker import (FIX, IGNORE_WARNINGS, REMOTE_DATA, SHOW_WARNINGS,
 _pytest_version = Version(pytest.__version__)
 PYTEST_GT_5 = _pytest_version > Version('5.9.9')
 PYTEST_GE_5_4 = _pytest_version >= Version('5.4')
-PYTEST_GE_7_0 = any([_pytest_version.is_devrelease,
+PYTEST_GE_7_0 = _pytest_version >= Version('7.0')
+PYTEST_GE_8_0 = any([_pytest_version.is_devrelease,
                      _pytest_version.is_prerelease,
-                     _pytest_version >= Version('7.0')])
+                     _pytest_version >= Version('8.0')])
 
 comment_characters = {
     '.txt': '#',
@@ -507,7 +508,11 @@ class DoctestPlus(object):
         Skip paths that match any of the doctest_norecursedirs patterns or
         if doctest_only is True then skip all regular test files (eg test_*.py).
         """
-        if PYTEST_GE_7_0:
+        if PYTEST_GE_8_0:
+            dirpath = Path(path).parent
+            collect_ignore = config._getconftest_pathlist("collect_ignore",
+                                                          path=dirpath)
+        elif PYTEST_GE_7_0:
             dirpath = Path(path).parent
             collect_ignore = config._getconftest_pathlist("collect_ignore",
                                                           path=dirpath,

